packages/components/tag/src/tag.ts

```diff
@@ -58,128 +65,105 @@ export class Tag extends ScopedElementsMixin(LitElement) {
+  /** @internal Emits when the tag is removed. */
   @event({ name: 'sl-remove' }) removeEvent!: EventEmitter<SlRemoveEvent>;
 
-  /** Whether the tag label is overflowing. */
-  #overflow = false;
+  /** The size of the tag. Defaults to `md`. */
+  @property({ reflect: true }) size?: TagSize;
 
   override connectedCallback(): void {
     super.connectedCallback();
 
-    this.setAttribute('id', 'tag-element');
-
-    if (this.readonly) {
-      this.setAttribute('aria-readonly', 'true');
+    if (!this.hasAttribute('tabindex')) {
+      this.setAttribute('tabindex', '0');
     }
-  }
 
-  override render(): TemplateResult {
-    return html`
-      <div class="wrapper" aria-describedby="tooltip-label" tabindex="0">
-        <div class="label">${this.label}</div>
-        ${this.removable && !this.readonly
-          ? html` <button
-              id="close-button"
-              aria-labelledby="tag-element"
-              aria-label=${msg('Remove')}
-              @mouseover=${this.#onMouseover}
-              @mouseout=${this.#onMouseover}
-              @click=${this.#onRemoveClick}
-              class="remove-button"
-              tabindex="-1"
-            >
-              <sl-icon name="xmark" .size=${this.size}></sl-icon>
-            </button>`
-          : nothing}
-      </div>
-      ${this.#overflow ? html`<sl-tooltip id="tooltip-label" position="bottom"> ${this.label} </sl-tooltip>` : nothing}
-    `;
+    this.#observer.observe(this);
   }
 
-  override firstUpdated(changes: PropertyValues<this>): void {
-    super.firstUpdated(changes);
+  override disconnectedCallback(): void {
+    this.#observer.disconnect();
+
+    if (this.#tooltip instanceof Tooltip) {
+      this.#tooltip?.remove();
+    } else if (this.#tooltip) {
+      this.#tooltip();
+    }
 
-    this.#checkOverflow();
-    this.requestUpdate();
+    this.#tooltip = undefined;
+
+    super.disconnectedCallback();
   }
 
   override updated(changes: PropertyValues<this>): void {
     super.updated(changes);
 
-    if (changes.has('readonly')) {
-      if (this.readonly) {
-        this.setAttribute('aria-readonly', 'true');
-      } else {
-        this.removeAttribute('aria-readonly');
-      }
-    }
-
     if (changes.has('disabled')) {
-      const wrapper = this.renderRoot.querySelector('.wrapper') as HTMLDivElement;
-      wrapper.setAttribute('tabindex', this.disabled ? '-1' : '0');
+      this.setAttribute('tabindex', this.disabled ? '-1' : '0');
     }
+  }
 
-    this.#checkOverflow();
+  override render(): TemplateResult {
+    return html`
+      <div class="wrapper" .tabIndex=${this.disabled ? -1 : 0}>
+        <slot @slotchange=${this.#onSlotChange}></slot>
+        ${this.removable
+          ? html`
+              <button @click=${this.#onRemove} aria-label=${msg(`Remove ${this.label}`)} ?disabled=${this.disabled}>
+                <sl-icon name="xmark" size=${ifDefined(this.size)}></sl-icon>
```

Comment:
I've just noticed that for `xmark` icon in Figma there is a `xs` size of icon used for `md` and `lg` size of tag,
 so 12x12 px:

<img width="1441" alt="image" src="https://github.com/user-attachments/assets/fee77bf9-48d0-4990-b525-b190afddac83">